src/main/java/org/folio/circulation/domain/RequestQueue.java

```diff
@@ -13,9 +13,11 @@
 public class RequestQueue {
 
   private List<Request> requests;
+  private final List<UpdatedRequestPair> updatedRequests;
```

Comment:
Why are a list of updated requests needed?

Reply:
It's a good question, Marc. I only tried to anticipate situations where a request can be updated several times or several requests can be updated during the lifecycle of the RequestQueue. Can I count on a contract that only one request can be updated once during the lifecycle of a particular RequestQueue? If so, it will greatly simplify the implementation.

Reply:
> I only tried to anticipate situations where a request can be updated several times or several requests can be updated during the lifecycle of the RequestQueue. Can I count on a contract that only one request can be updated once during the lifecycle of a particular RequestQueue? If so, it will greatly simplify the implementation.

No, it isn't guaranteed that only one request will be updated during a particular request to a circulation endpoint.

I'm not sure I really understand the approach taken here. Rather than trying to anticipate situations, why not work backwards from the situations that the circulation log has been intended to capture?

What information about requests changing is the circulation log trying to capture? And which are specifically included in this change?

Reply:
Actually, we need:
**new request status
type
pickup service point/address type
fulfilment preference
old request status
expiration date
position in queue**
for each request updating during check-in/check-out flow.

Reply:
As far as it concerns approach. I'm going to populate requests before/after updating (including case when several requests can be updated during check-in/check-out flow - may be I'm incorrect here) and send this information to mod-audit for further processing.

Reply:
Is the circulation log only concerned with requests that have been updated during a check in / check out?

Reply:
Yes, it's correct for this PR.
But we also have separate task to log all requests statuses changing.